src/sqlfluff/core/parser/segments/base.py

```diff
@@ -885,25 +885,24 @@ def select_children(
                 buff.append(seg)
         return buff
 
-    def recursive_crawl(self, *seg_type, recurse_into=True):
-        """Recursively crawl for segments of a given type.
-
-        Args:
-            seg_type: :obj:`str`: one or more type of segment
-                to look for.
-            recurse_into: :obj:`bool`: When an element of type "seg_type" is
-                found, whether to recurse into it.
-        """
+    def recursive_crawl(
+        self, *seg_type: str, recurse_into: bool = True, all_descendants: bool = False
```

Comment:
Added new parameter `all_descendants` to get _all_ the segments, rather than filtering by type.